src/exec/noexec.rs

```diff
@@ -281,11 +361,41 @@ pub(crate) fn add_noexec_filter(command: &mut Command) -> SpawnNoexecHandler {
             // SAFETY: seccomp_data can be safely zero-initialized.
             let dummy: seccomp_data = zeroed();
             let nr_offset = (&dummy.nr) as *const _ as usize - &dummy as *const _ as usize;
+            let arch_offset = (&dummy.arch) as *const _ as usize - &dummy as *const _ as usize;
 
             // SAFETY: libc unnecessarily marks these functions as unsafe
-            let exec_filter: [sock_filter; 5] = [
+            #[rustfmt::skip]
+            let exec_filter = [
+                // Load architecture number into the accumulator
+                BPF_STMT((BPF_LD | BPF_ABS) as _, arch_offset as _),
+                // Check if we are any of the recognized architectures
+                BPF_JUMP((BPF_JMP | BPF_JEQ | BPF_K) as _, HOST_ARCH as _, 7, 0),
+                BPF_JUMP((BPF_JMP | BPF_JEQ | BPF_K) as _, GUEST_ARCH as _, 1, 0),
+                // Not a recognized architecture, forbid all syscalls
+                BPF_STMT((BPF_RET | BPF_K) as _, SECCOMP_RET_USER_NOTIF as _),
```

Comment:
```suggestion
                BPF_STMT((BPF_RET | BPF_K) as _, SECCOMP_RET_KILL_PROCESS as _),
```

there is nothing useful that the process can do anyway.